test/e2eprovider/server/server_test.go

```diff
@@ -0,0 +1,141 @@
+// +build e2e
+
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+	"testing"
+
+	"sigs.k8s.io/secrets-store-csi-driver/provider/v1alpha1"
+)
+
+var (
+	testMockServer *Server
+	tempDir        = os.TempDir()
+)
+
+func TestMain(m *testing.M) {
+	setup()
+	exitCode := m.Run()
+	teardown()
+	os.Exit(exitCode)
+}
+
+func setup() {
+	var err error
+
+	testMockServer, err = NewE2EProviderServer(fmt.Sprintf("unix://%s/%s", tempDir, "e2e-provider.sock"))
+	if err != nil {
+		panic(err)
+	}
+}
+
+func teardown() {
+	testMockServer.Stop()
+	os.RemoveAll(tempDir)
+}
+
+func TestMockServer(t *testing.T) {
+	cases := []struct {
+		name string
+	}{
+		{
+			name: "start",
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := testMockServer.Start()
+			if err != nil {
+				t.Errorf("Did not expect error on server start: %v", err)
+			}
+		})
+	}
+}
+
+func TestMount(t *testing.T) {
+	cases := []struct {
+		name    string
+		input   *v1alpha1.MountRequest
+		want    *v1alpha1.MountResponse
+		wantErr error
+	}{
+		{
+			"Parse static secrets",
```

Comment:
Maybe also add a test here for validating the rotation logic?

Reply:
done